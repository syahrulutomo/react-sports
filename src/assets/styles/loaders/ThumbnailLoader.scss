.thumbnail-loader {
  margin-right: $space;
  width: 120px;
  height: 80px;
  background-color: $theme-ui-colors-grey3;
  border-radius: 4px;

  &::after {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    transform: translateX(-100%);
    background-image: linear-gradient(
      90deg,
      rgba($theme-ui-colors-white, 0) 0,
      rgba($theme-ui-colors-grey1, 0.2) 20%,
      rgba($theme-ui-colors-white, 0.3) 60%,
      rgba($theme-ui-colors-white, 0)
    );
    animation: shimmer 2s infinite;
    content: '';
  }

  @keyframes shimmer {
    100% {
      transform: translateX(100%);
    }
  }

  &:last-of-type {
    margin-right: 0;
  }

}