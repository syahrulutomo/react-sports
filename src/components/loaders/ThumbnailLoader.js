import React from 'react';
import PropTypes from 'prop-types';

export const ThumbnailLoader = (props) => {
  const { style } = props;
  return (
    <div className="thumbnail-loader" style={style}/>
  );
}

ThumbnailLoader.propTypes = {
  style: PropTypes.object
};

