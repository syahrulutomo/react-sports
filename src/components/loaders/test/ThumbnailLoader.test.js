import React from 'react';
import {cleanup} from '@testing-library/react';
import { ThumbnailLoader } from '../ThumbnailLoader'
import ShallowRenderer from 'react-test-renderer/shallow';

afterEach(cleanup)

it('renders correctly ThumbnailLoader component', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<ThumbnailLoader />);
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});