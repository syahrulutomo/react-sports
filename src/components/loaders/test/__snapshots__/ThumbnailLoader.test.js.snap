// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly ThumbnailLoader component 1`] = `
<div
  className="thumbnail-loader"
/>
`;
